docker/package/baking_wizard.py

```diff
@@ -0,0 +1,303 @@
+            validator=[required_field_validator, filepath_validator],
+            reqs=lambda vs: vs["mode"] == "baker" and vs["key_import_mode"] == "json",
+        ),
+    ]
+
+    for step in steps:
+        if not step.reqs(values):
+            continue
+
+        readline.set_completer(step.completer)
+
+        validated = False
+        while not validated:
+            try:
+                answer = input(step.prompt).strip()
+            except:
+                print("Exiting Tezos Baking Wizard")
+                return None
+
+            if answer.lower() in ["quit", "exit"]:
+                print("Exiting Tezos Baking Wizard")
+                return None
+            elif answer.lower() == "help":
+                print(step.help)
+            else:
+                if not answer and step.default is not None:
+                    answer = step.default
+
+                try:
+                    step.validate(answer)
+                except ValueError as e:
+                    print("Validation error: ", str(e))
+                else:
+                    validated = True
+                    values[step.id] = answer
+
+    return values
+
+
+class Setup:
+    def __init__(self, config):
+        self.config = config
+
+    # Bootstrapping tezos-node
+    def bootstrap_node(self):
+        if self.config["snapshot"]:
+            proc_call(
+                "sudo -u tezos tezos-node-"
+                + self.config["network"]
+                + " snapshot import "
+                + self.config["snapshot"]
+            )
+
+        proc_call(
+            "sudo systemctl start tezos-node-" + self.config["network"] + ".service"
+        )
+
+        return_code = None
+        while return_code != 0:
+            os.system("sleep 5")
+            return_code = subprocess.call(
+                shlex.split("sudo -u tezos tezos-client bootstrapped")
+            )
+
+    # Importing the baker key
+    def import_baker_key(self):
+        if self.config["key_import_mode"] == "secret-key":
+            proc_call(
+                "sudo -u tezos tezos-client import secret key baker"
+                + self.config["secret_key"]
+            )
+        elif self.config["key_import_mode"] == "json":
+            proc_call(
+                "sudo -u tezos tezos-client activate account baker with"
+                + self.config["json_filepath"]
+            )
+
+        else:
+            print()
+            input("Please open the Tezos Wallet app on your ledger, then hit Enter.")
+            proc_call(
+                "sudo -u tezos tezos-client import secret key baker"
+                + self.config["ledger_url"]
+            )
+
+            print()
+            input("Please open the Tezos Baking app on your ledger, then hit Enter.")
+            proc_call("sudo -u tezos tezos-client setup ledger to bake for baker")
+
+    def register_baker(self):
+        proc_call("sudo -u tezos tezos-client register key baker as delegate")
```

Comment:
AFAIR, baker registration also requires the Baking app to be open in case the baker key was imported from the ledger

Reply:
:thinking: It's not mentioned in the baking guide at any rate. If so, then that's something to fix there too

Reply:
I am not sure this is true. Registration fee must be paid to register, and for that transaction must be signed (of type "transfer", I think), so if anything I would expect tezos wallet app should be running

Reply:
I left it commented at this moment, but I'd really like to get confirmation if there should or shouldn't be a prompt for any app open on the ledger.

Reply:
I confirm that a response on the ledger is required to run `tezos-client register key baker as delegate`. Both Baking and Wallet app can be used to confirm this operation